internal/pkg/sessions/cookie_store.go

```diff
@@ -81,13 +80,14 @@ func NewCookieStore(cookieName string, optFuncs ...func(*CookieStore) error) (*C
 
 func (s *CookieStore) makeCookie(req *http.Request, name string, value string, expiration time.Duration, now time.Time) *http.Cookie {
 	logger := log.NewLogEntry()
-	domain := req.Host
-	if h, _, err := net.SplitHostPort(domain); err == nil {
-		domain = h
-	}
+
+	// To avoid subdomains being inadvertently included, omit the domain attribute unless
+	// it's explicitly set in s.CookieDomain.
+	// https://datatracker.ietf.org/doc/html/rfc6265#section-4.1.2.3
+	domain := ""
 	if s.CookieDomain != "" {
```

Comment:
I'm not actually finding any cases (other than test code) where `CookieDomain` (via the `CookieConfig.Domain` value) would be set -- am I missing something?

Reply:
Right, as far as I understand the only other time it's set is if it's explicitly set as a configuration variable.

As you've mentioned, that would be through the `CookieConfig.Domain` configuration variable, which is then set as part of the `CookieStore` here (and the equivalent in `sso_auth`):
https://github.com/buzzfeed/sso/blob/9c912b1e1d6e3cc235795a69240ce167c27954ad/internal/proxy/options.go#L30-L38


I suppose we could simply set the cookie domain attribute to always match the `CookieConfig.Domain` value, seeing as it will either be explicitly set, in which case we want to use it, or the zero value of `""` anyway.

Reply:
We're on the same page. I think what you've got now makes a lot of sense but maybe ought to be documented over [here](https://github.com/buzzfeed/sso/blob/main/docs/sso_authenticator_config.md#L14), to make very clear the behavior of omitting that configuration value or setting it empty?

Reply:
I made a small adjustment to the documentation there already, but I think you're right -- this cookie behavior (even outside of this application) isn't necessarily what one might expect, so making that even more explicit and clear is 👍

Reply:
I've pushed up a change with some additional changes to the documentation.